reference/5.0/Microsoft.PowerShell.Core/Register-ArgumentCompleter.md

```diff
@@ -113,9 +191,42 @@ Accept wildcard characters: False
 
 ### -ScriptBlock
 
-Specifies the commands to run.
-Enclose the commands in braces ( { } ) to create a script block.
-This parameter is required.
+Specifies the commands to run to perform tab completion. The script block you provide should return
+the values complete the input. The script block must unroll the values using the pipeline
+(`ForEach-Object`, `Where-Object`, etc.), or another suitable method. Returning an array of values
+causes PowerShell to treat the entire array as **one** tab completion value.
+
+The script block should also accept the following parameters in the order specified below. The names
+of the parameters are not important because PowerShell passes in the values *positionally*.
+
+- `$commandName` (Position 0) - This parameter is set to the name of the
+  command for which the script block is providing tab completion.
+- `$parameterName` (Position 1) - This parameter is set to the parameter
+  whose value requires tab completion.
+- `$wordToComplete` (Position 2) - This parameter is set to value the user has
+  provided before they pressed `<TAB>`. Your script block should use this value
+  to determine tab completion values.
+- `$commandAst` (Position 3) - This parameter is set to the Abstract Syntax
+  Tree (AST) for the current input line. For more information, see
+  [Ast Class](/dotnet/api/system.management.automation.language.ast).
+- `$fakeBoundParameter` (Position 4) - This parameter is set to a hashtable
+  containing the `$PSBoundParameters` for the cmdlet, before the user pressed
+  `<TAB>`. For more information, see [about_Automatic_Variables](./About/about_Automatic_Variables.md).
+
+When you specify the **Native** parameter, the script block should take the following parameters in
```

Comment:
My experience is that you also get the following parameters when you don't specify the `-ParameterName` parameter (even if you haven't specified -Native).

Reply:
@rkeithhill So is there something that should be changed in the docs? How would you word it?

Reply:
It is not when you specify `-Native`, it is when you don't specify `-ParameterName`.  Register-ArgumentCompleter has these two parameter sets:
```
Register-ArgumentCompleter -CommandName <string[]> -ScriptBlock <scriptblock> [-Native] [<CommonParameters>]

Register-ArgumentCompleter -ParameterName <string> -ScriptBlock <scriptblock> [-CommandName <string[]>] [<CommonParameters>]
```
If you specify `-ParameterName` you get the second parameter set and five parameters get sent to the ScriptBlock.  If you do not specify `-ParameterName`, then you get the first parameter set and you only get three parameters.  

A quick experiment verifies this:
```
function Test-Command([CmdletBinding][Parameter()][string]$Name) {$Name} 
$argCompleter = {$OFS=", "; $global:COMPLETER = "$($args | % GetType | Out-String)"}
Register-ArgumentCompleter -CommandName Test-Command -ScriptBlock $argCompleter
Test-Command <tab>
```
Now inspect the variable the scriptblock created:
```
PS> $COMPLETER                                                                                    
IsPublic IsSerial Name                                     BaseType
-------- -------- ----                                     --------
True     True     String                                   System.Object
True     False    CommandAst                               System.Management.Automation.Language.CommandBaseAst       
True     True     Int32                                    System.ValueType
```
So long story short, you could simply change line 216 to say `When you do not specify the **ParameterName** parameter, ...`